### DISCUSSION TITLE
Application permission

### DISCUSSION THREAD

**Discussion by juanneres** [pnp/powershell#3514] [General]
Is it possible to use PnP PowerShell with application permissions so that I can utilize tenantID, clientID, and Secret for authentication instead of using a username and password?

My requirement is to monitor SharePoint metrics for various clients, and I use MFA for all of them. I intend to automate tasks, but it becomes unfeasible with MFA.

**Comment by AndersRask** [comment]
Yes, but would recommend using certificates instead of secret for most scenarios, as secrets use legacy authentication with limited use in the module.

Read more here 
https://pnp.github.io/powershell/articles/authentication.html
https://pnp.github.io/powershell/articles/connecting.html
